x-pack/plugins/spaces/server/new_platform/spaces_service/spaces_service.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { map, take } from 'rxjs/operators';
+import { Observable, Subscription, combineLatest } from 'rxjs';
+import { Legacy } from 'kibana';
+import {
+  Logger,
+  ElasticsearchServiceSetup,
+  HttpServiceSetup,
+  KibanaRequest,
+  SavedObjectsService,
+} from 'src/core/server';
+import { OptionalPlugin } from '../../../../../server/lib/optional_plugin';
+import { DEFAULT_SPACE_ID } from '../../../common/constants';
+import { SecurityPlugin } from '../../../../security';
+import { SpacesClient } from '../../lib/spaces_client';
+import { getSpaceIdFromPath } from '../../lib/spaces_url_parser';
+import { SpacesConfigType } from '../config';
+
+type RequestFacade = KibanaRequest | Legacy.Request;
+
+export interface SpacesServiceSetup {
+  scopedClient(request: RequestFacade): Promise<SpacesClient>;
```

Comment:
a bit misleading. `SpacesClient` cannot work with KibanaRequest, thus we can scope only to Legacy.Request